src/lib/selectors.ts

```diff
@@ -0,0 +1,623 @@
+      if (!sdk) {
+        reject(new Error('Sdk not found'));
+      }
+      if (!(sdk?.connect && typeof sdk.connect === 'function')) {
+        reject(new Error('Not found the function "connect'));
+      }
+      if (!accessToken) {
+        sdk.connect(userId)
+          .then((res) => resolve(res))
+          .catch((err) => reject(err));
+      } else {
+        sdk.connect(userId, accessToken)
+          .then((res) => resolve(res))
+          .catch((err) => reject(err));
+      }
+    })
+  )
+);
+
+/**
+ * const disconnect = selectors.disconnect(state);
+ * disconnect()
+ *  .then(() => {})
+ *  .catch((error) => {})
+ */
+export const disconnect = (state: SendBirdState) => (
+  (): Promise<void> => (
+    new Promise((resolve, reject) => {
+      const sdk = getSdk(state);
+      if (!sdk) {
+        reject(new Error('Sdk not found'));
+      }
+      if (!(sdk.disconnect && typeof sdk.disconnect === 'function')) {
+        reject(new Error('Not found the function "disconnect'));
+      }
+      sdk.disconnect()
+        .then((res) => resolve(res))
+        .catch((err) => reject(err));
+    })
+  )
+);
+
+/**
+ * const updateUserInfo = selectors.updateUserInfo(state);
+ * updateUserInfo('new-nickname', 'new-profile-url')
+ *  .then((user) => {})
+ *  .catch((error) => {})
+ */
+export const updateUserInfo = (state: SendBirdState) => (
+  (nickname: string, profileUrl: string): Promise<User> => (
+    new Promise((resolve, reject) => {
+      const sdk = getSdk(state);
+      if (!sdk) {
+        reject(new Error('Sdk not found'));
+      }
+      if (!(sdk.updateCurrentUserInfo && typeof sdk.updateCurrentUserInfo === 'function')) {
+        reject(new Error('Not found the function "updateCurrentUserInfo"'))
+      }
+      sdk.updateCurrentUserInfo({ nickname, profileUrl } as UserUpdateParams)
+        .then((res) => resolve(res))
+        .catch((err) => reject(err));
+    })
+  )
+);
+
+// 3. Channel
+
+/**
+ * const createGroupChannel = selectors.createGroupChannel(state);
+ * createGroupChannel(channelParams: GroupChannelCreateParams)
+ *  .then((channel) => {})
+ *  .catch((error) => {})
+ */
+export const createGroupChannel = (state: SendBirdState) => (
+  (params: GroupChannelCreateParams): Promise<GroupChannel> => (
+    new Promise((resolve, reject) => {
+      const sdk = getSdk(state);
+      // const pubSub = getPubSub(state);
+      if (!sdk) {
+        reject(new Error('Sdk not found'));
+      }
+      if (!sdk.groupChannel) {
+        reject(new Error('Not found GroupChannelModule'));
+      }
+      if (!(sdk.groupChannel.createChannel && typeof sdk.groupChannel.createChannel === 'function')) {
+        reject(new Error('Not found the function "createChannel"'))
+      }
+      sdk.groupChannel.createChannel(params)
+        .then((channel) => {
+          resolve(channel);
+          // pubSub.publish(
+          //   topics.CREATE_CHANNEL,
+          //   { channel },
+          // );
```

Comment:
Dotn we need to publish this?

Reply:
oops, I forgot to make it return to normal code